test/jake/jakelib/migrate-test.js

```diff
@@ -151,6 +151,123 @@ test('migrate:rollback prints verbose logs', (temp) => {
   });
 });
 
+test('migrate:rollback --all rolls back all completed migrations', (temp) => {
+  return (
+    new Promise((resolve, reject) => {
+      fs.writeFile(
+        `${temp}/migrations/000_create_users_table.js`,
+        `
+        exports.up = (knex) => knex.schema
+          .createTable('users', (table) => {
+            table.integer('id');
+          });
+
+        exports.down = (knex) => knex.schema.dropTable('users');
+      `,
+        (err) => (err ? reject(err) : resolve())
+      );
+    })
+      .then(() => {
+        return assertExec(
+          `${KNEX} migrate:latest \
+      --client=sqlite3 \
+      --connection=${temp}/db \
+      --migrations-directory=${temp}/migrations`,
+          'create_users_table'
+        );
+      })
+
+      // .then(() => {
+      //   console.log(fs.readdirSync(`${temp}/migrations`));
+
+      //   return new sqlite3.Database(temp + '/db')
+      // })
+      // .then((db) => {
+      //   return new Promise((resolve, reject) => {
+      //     db.get('SELECT * FROM knex_migrations', function(err, row) {
+      //       console.log(err);
+      //       console.log(row)
+
+      //       return err ? reject(err) : resolve(row);
+      //     })
+      //   });
+      // })
+
+      .then(() => {
+        return new Promise((resolve, reject) => {
+          fs.writeFile(
+            `${temp}/migrations/001_add_columns_to_user_table.js`,
+            `
+          exports.up = (knex) => knex.schema
+            .table('users', (table) => {
+              table.string('name');
+            });
+
+          exports.down = (knex) => knex.schema
+            .table('users', (table) => {
+              table.dropColumn('name');
+            });
+        `,
+            (err) => (err ? reject(err) : resolve())
+          );
+        });
+      })
+      .then(() => {
+        return assertExec(
+          `${KNEX} migrate:latest \
+      --client=sqlite3 \
+      --connection=${temp}/db \
+      --migrations-directory=${temp}/migrations`,
+          'add_columns_to_user_table'
+        );
+      })
+
+      // .then(() => {
+      //   console.log(fs.readdirSync(`${temp}/migrations`));
+
+      //   return new sqlite3.Database(temp + '/db')
+      // })
+      // .then((db) => {
+      //   return new Promise((resolve, reject) => {
+      //     db.get('SELECT * FROM knex_migrations', function(err, row) {
+      //       console.log(err);
+      //       console.log(row)
+
+      //       return err ? reject(err) : resolve(row);
+      //     })
+      //   });
+      // })
+
+      .then(() => {
+        return new Promise((resolve, reject) => {
+          fs.writeFile(
```

Comment:
if after using writeFileSync and ensuring that files indeed exist at the moment you are running migrations, you still are only running the very first migration, I would recommend checking if knexfile.js has correct path to migrations directory, could be that it's looking in a different place.

Reply:
Maybe I'm way off base here but why would it only run the very first migration even if there are more migrations in that directory? Shouldn't `migration:latest` run all migrations that haven't been run yet?

Reply:
It should run all of them, which is why I'm slightly suspicious about async write not completing when expected or knex reading from wrong directory.

Reply:
That makes sense. I'll investigate